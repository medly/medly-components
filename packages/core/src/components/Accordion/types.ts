export type StaticProps = {
    Header: React.FC;
    Content: React.FC;
    Context: React.Context<AccordionContextType>;
};

export type AccordionContextType = [
    boolean /** Current active state */,
    React.Dispatch<React.SetStateAction<boolean>> /** Function to be called to toggle the active state */
];
