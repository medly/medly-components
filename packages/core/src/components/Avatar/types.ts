import { AvatarSizes } from '@medly-components/theme';
import { HTMLProps, Omit } from '@medly-components/utils';

export interface AvatarProps extends Omit<HTMLProps<HTMLDivElement>, 'size'> {
    /** Size of the avatar */
    size?: AvatarSizes;
    /** Text color */
    textColor?: string;
    /** Background color */
    bgColor?: string;
    /** Hover Text color */
    hoverTextColor?: string;
    /** Hover Background color */
    hoverBgColor?: string;
    /** Hover text shadow color*/
    hoverTextShadowColor?: string;
    /** Hover image shadow color */
    hoverImgShadowColor?: string;
    /** Pressed Background color */
    pressedBgColor?: string;
    /** Pressed Text color */
    pressedTextColor?: string;
}

export interface StyledProps extends AvatarProps {
    /** To check children is image */
    isImage?: boolean;
}
