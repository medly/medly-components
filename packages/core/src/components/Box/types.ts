import { FontVariants, Theme } from '@medly-components/theme';
import { HTMLProps } from '@medly-components/utils';

export type BoxProps = Omit<HTMLProps<HTMLDivElement>, 'size'> & { theme?: Theme } & {
    /** To be used to render any component as Button */
    as?: keyof JSX.IntrinsicElements | React.ComponentType<any>;
    /** Padding along x-axis */
    px?: number;
    /** Padding along y-axis */
    py?: number;
    /** Margin along x-axis */
    mx?: number;
    /** Margin along y-axis */
    my?: number;
    /** Background color */
    bg?: string;
    /** To be used to specify color of the text inside component */
    color?: string;
    /** To be used to specify width of the component */
    width?: string;
    /** To be used to specify height of the component */
    height?: string;
    /** To be used to specify border color, borderWidth is also required for this */
    borderColor?: string;
    /** To be used to specify border width, borderColor is also required for this */
    borderWidth?: number;
    /** To be used to change border-radius */
    borderRadius?: number;
    /** To be used to specify the display type of component */
    display?: 'flex' | 'block' | 'none' | 'inline-flex' | 'inline-block' | 'table' | 'table-row' | 'list-item' | 'initial';
    /** Set it true to show loading state */
    isLoading?: boolean;
    /** Set font variant for text inside component */
    fontVariant?: FontVariants;
    /** Set shadow for the box component */
    shadowSize?: 'S' | 'M' | 'L' | 'XL';
    /** Set shadow color for the box component */
    shadowColor?: string;
};
