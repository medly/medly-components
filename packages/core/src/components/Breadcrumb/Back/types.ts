import { HTMLProps } from '@medly-components/utils';

export type BreadcrumbBackProps = HTMLProps<HTMLLIElement> & {
    as?: keyof JSX.IntrinsicElements | React.ComponentType<any>;
};
