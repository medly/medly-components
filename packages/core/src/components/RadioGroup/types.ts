import { FontVariants, FontWeights, RadioSizes } from '@medly-components/theme';
import { HTMLProps, Omit } from '@medly-components/utils';

export interface RadioGroupProps extends Omit<HTMLProps<HTMLDivElement>, 'size' | 'onChange' | 'type' | 'ref'> {
    /** Selected radio value */
    value?: string;
    /** Name to be given to all radio option as html attribute */
    name?: string;
    /** Function to be called on selecting any radio */
    onChange?: (value: any) => void;
    /** Options */
    options: { value: any; label: string; disabled?: boolean }[];
    /** Function will be called with the input value on Blur event */
    validator?: (val: any) => string;
    /** Takes full width of the parent component */
    fullWidth?: boolean;
    /** Radio group label */
    label?: string;
    /** Use when required in any form */
    required?: boolean;
    /** Radio size */
    size?: RadioSizes;
    /** Use to disable radio selection */
    disabled?: boolean;
    /** Label variant */
    labelVariant?: FontVariants;
    /** Label weight */
    labelWeight?: FontWeights;
    /** Error text for radioGroup */
    errorText?: string;
    /** Helper text for radioGroup */
    helperText?: string;
    /** Use this prop to render option in multiple columns */
    columns?: number;
}
