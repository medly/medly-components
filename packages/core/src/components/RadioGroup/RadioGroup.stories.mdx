import { RadioGroup } from './RadioGroup';
import { boolean, number, select, text, withKnobs } from '@storybook/addon-knobs';
import { Preview, Story, Meta, Props } from '@storybook/addon-docs/blocks';
import { FormWithRadioGroup, sizes } from './RadioGroup.stories';
import { useState } from 'react';

<Meta
    title="Core"
    component={RadioGroup}
    parameters={{
        jest: ['RadioGroup.test.tsx'],
        design: {
            type: 'figma',
            url: 'https://www.figma.com/file/vA4EQdxl0d0ETtYpw2uSjW/Medly-Pattern-Library-B2B-Internal?node-id=239%3A0'
        }
    }}
/>

# RadioGroup

The `RadioGroup` is a wrapper used to group two or more Radio components. It is a controlled react component. It provides an easier API and proper keyboard accessibility to the group.
Handle the state as mentioned below.

```tsx
const [value, setValue] = useState('female');

return (
    <RadioGroup
        label="Gender"
        name="gender"
        value={value}
        onChange={setValue}
        options={[
            { value: 'female', label: 'Female' },
            { value: 'male', label: 'Male' }
        ]}
    />
);
```

<Preview withToolbar>
    <Story name="RadioGroup" decorators={[withKnobs]}>
        {() => {
            const [value, setValue] = useState('female');
            return (
                <RadioGroup
                    id="gender"
                    value={value}
                    onChange={setValue}
                    disabled={boolean('Disabled', false)}
                    required={boolean('Required', false)}
                    fullWidth={boolean('Full Width', false)}
                    label={text('Label', 'Gender')}
                    size={select('Size', sizes, 'S')}
                    errorText={text('Error Text', '')}
                    helperText={text('Helper Text', 'Helper Text')}
                    columns={number('Columns', 1)}
                    options={[
                        { value: 'female', label: 'Female' },
                        { value: 'male', label: 'Male' }
                    ]}
                />
            );
        }}
    </Story>
</Preview>

## Validations

1. For now, by default, the RadioGroup component can only display the error message on the form submit when you pass the required prop.

```tsx
() => {
    const [value, setValue] = useState(),
        handleSubmit = e => e.preventDefault();
    return (
        <form onSubmit={handleSubmit}>
            <RadioGroup
                required
                label="Gender"
                name="gender"
                value={value}
                onChange={setValue}
                options={[
                    { value: 'female', label: 'Female' },
                    { value: 'male', label: 'Male' }
                ]}
            />
            <Button type="submit">Submit</Button>
        </form>
    );
};
```

<Preview withToolbar>
    <FormWithRadioGroup />
</Preview>

2. If you pass the validator function, it will be called with the latest selected value. If needed, you can return the desired error message.

```tsx
validator={val => (!val ? 'Please select any one company' : '')}
```

<Preview withToolbar>
    <FormWithRadioGroup
        label="Laptop"
        options={[
            { value: 'apple', label: 'Apple' },
            { value: 'dell', label: 'Dell' }
        ]}
        validator={val => (!val ? 'Please select any one company' : '')}
    />
</Preview>

## Props

This component offers the following props mentioned in the table below:

<Props of={RadioGroup} />

## Theme

This component offers the same theme as Radio. To view the same click [here](http://www.medlycomponents.com/?path=/docs/core--radio#radiotheme)

