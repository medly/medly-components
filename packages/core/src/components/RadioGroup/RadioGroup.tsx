import { useCombinedRefs, WithStyle } from '@medly-components/utils';
import type { ChangeEvent, FC, FocusEvent, FormEvent } from 'react';
import { forwardRef, memo, useCallback, useMemo, useRef, useState } from 'react';
import Radio from '../Radio';
import { SelectorGroup } from '../Selectors';
import { RadioGroupProps } from './types';
const Component: FC<RadioGroupProps> = memo(
    forwardRef((props, ref) => {
        const {
            id,
            size,
            value,
            name,
            label,
            disabled,
            required,
            options,
            labelVariant,
            labelWeight,
            errorText,
            helperText,
            columns,
            validator,
            onChange,
            onBlur,
            onInvalid,
            ...wrapperProps
        } = props;

        const [builtInErrorMessage, setErrorMessage] = useState(''),
            radioGroupId = useMemo(() => id || name, [id, name]),
            radioGroupRef = useCombinedRefs<HTMLDivElement>(ref, useRef(null)),
            hasError = useMemo(() => !!errorText || !!builtInErrorMessage, [errorText, builtInErrorMessage]),
            hasHelperOrErrorText = useMemo(() => !!(errorText || helperText), [errorText, helperText]);

        const validate = useCallback(
            (selectedValue: string, validationMessage?: string) =>
                setErrorMessage((validator && validator(selectedValue)) || validationMessage || ''),
            [validator]
        );

        const handleOnInvalid = useCallback(
                (event: FormEvent<HTMLInputElement>) => {
                    validate(value || '', (event.target as HTMLInputElement).validationMessage);
                    onInvalid && onInvalid(event);
                },
                [validate, onInvalid, value]
            ),
            handleOnBlur = useCallback(
                (event: FocusEvent<HTMLDivElement>) => {
                    const { currentTarget } = event,
                        target = event.target as HTMLInputElement;
                    setTimeout(() => !currentTarget.contains(document.activeElement) && validate(value || '', target.validationMessage), 0);
                    onBlur && onBlur(event);
                },
                [validate, onBlur, value]
            ),
            handleOnChange = useCallback(
                (event: ChangeEvent<HTMLInputElement>) => {
                    const { value } = event.target;
                    validate(value);
                    onChange && onChange(value);
                },
                [validate]
            );

        return (
            <SelectorGroup.Wrapper
                id={`${radioGroupId}-wrapper`}
                ref={radioGroupRef}
                type="radio"
                aria-describedby={`${radioGroupId}-helper-text`}
                onBlur={handleOnBlur}
                onInvalid={handleOnInvalid}
                {...{ ...wrapperProps, hasHelperOrErrorText }}
            >
                {label && (
                    <SelectorGroup.Label
                        type="radio"
                        hasError={hasError}
                        id={`${radioGroupId}-label`}
                        textVariant={labelVariant}
                        textWeight={labelWeight}
                        {...{ required, disabled }}
                    >
                        {label}
                    </SelectorGroup.Label>
                )}
                {(hasError || helperText) && (
                    <SelectorGroup.HelperText id={`${radioGroupId}-helper-text`} type="radio" {...{ disabled, hasError }}>
                        {errorText || builtInErrorMessage || helperText}
                    </SelectorGroup.HelperText>
                )}
                <SelectorGroup.Options id={`${radioGroupId}-options`} columns={columns}>
                    {options.map(option => (
                        <Radio
                            id={`${option.label}-${radioGroupId}`}
                            key={option.value}
                            onChange={handleOnChange}
                            checked={option.value === value}
                            {...{ ...option, disabled: disabled || option.disabled, name, size, required, hasError }}
                        />
                    ))}
                </SelectorGroup.Options>
            </SelectorGroup.Wrapper>
        );
    })
);

Component.displayName = 'RadioGroup';
Component.defaultProps = {
    columns: 1,
    fullWidth: true,
    labelWeight: 'Medium',
    labelVariant: 'body1'
};
export const RadioGroup: FC<RadioGroupProps> & WithStyle = Object.assign(Component, { Style: SelectorGroup.Wrapper });
