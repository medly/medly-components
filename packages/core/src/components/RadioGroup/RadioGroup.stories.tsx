import React, { useState } from 'react';
import Button from '../Button';
import { RadioGroup } from './RadioGroup';
import { RadioGroupProps } from './types';

export const sizes: RadioGroupProps['size'][] = ['XS', 'S', 'M', 'L', 'XL'];

export const FormWithRadioGroup: React.FC = (props: RadioGroupProps) => {
    const [value, setValue] = useState(),
        handleSubmit = (e: React.FormEvent) => e.preventDefault();

    return (
        <form onSubmit={handleSubmit}>
            <RadioGroup {...props} required name="gender" value={value} onChange={setValue} />
            <Button type="submit">Submit</Button>
        </form>
    );
};
FormWithRadioGroup.defaultProps = {
    label: 'Bikes',
    options: [
        { value: 'bajaj', label: 'Bajaj' },
        { value: 'apache', label: 'Apache' }
    ]
};
