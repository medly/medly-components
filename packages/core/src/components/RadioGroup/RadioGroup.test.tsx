import { fireEvent, render, screen } from '@test-utils';
import userEvent from '@testing-library/user-event';
import { RadioGroup } from './RadioGroup';
import { RadioGroupProps } from './types';

const renderer = ({
    label = 'Gender',
    name = 'gender',
    helperText = 'Helper Text',
    errorText = '',
    onChange = jest.fn(),
    required = false,
    validator,
    onInvalid = jest.fn(),
    onBlur = jest.fn(),
    disabled = false,
    options = [
        { value: 'female', label: 'Female' },
        { value: 'male', label: 'Male' }
    ]
}: Partial<RadioGroupProps>) =>
    render(<RadioGroup {...{ required, disabled, label, name, onInvalid, onBlur, validator, helperText, options, errorText, onChange }} />);

describe('Radio Group', () => {
    it('should render properly', () => {
        const { container } = renderer({});
        expect(container).toMatchSnapshot();
    });

    it('should render disabled state properly', () => {
        const { container } = renderer({ disabled: true });
        expect(container).toMatchSnapshot();
    });

    it('should be able to render error text', () => {
        renderer({ errorText: 'Something went wrong' });
        const errorText = screen.getByText('Something went wrong');
        expect(errorText).toBeInTheDocument();
        expect(errorText).toHaveStyle(`color: rgb(215, 58, 67)`);
    });

    it('should call onChange prop with selected option', () => {
        const mockOnChange = jest.fn();
        renderer({ onChange: mockOnChange });
        userEvent.click(screen.getByText('Male'));
        expect(mockOnChange).toBeCalledWith('male');
    });

    it('should show the html5 form validation on error', () => {
        renderer({ required: true });
        fireEvent.invalid(screen.getByRole('radio', { name: 'Female' }));
        expect(screen.getByText('Constraints not satisfied')).toBeInTheDocument();
    });

    it('should call validator function on blur', async () => {
        const mockOnBlur = jest.fn();
        renderer({
            onBlur: mockOnBlur,
            validator: () => 'Please select any one gender'
        });
        fireEvent.blur(screen.getByRole('radio', { name: 'Female' }));
        expect(await screen.findByText('Please select any one gender')).toBeInTheDocument();
    });
});
