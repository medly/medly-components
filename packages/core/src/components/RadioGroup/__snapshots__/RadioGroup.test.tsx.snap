// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Group should render disabled state properly 1`] = `
.c1 {
  margin: 0;
  color: inherit;
  font-size: 1.6rem;
  font-weight: 600;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.6rem;
  text-align: initial;
  display: inline-block;
}

.c6 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
  display: inline-block;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: not-allowed;
  color: #98A7B7;
}

.c7 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: not-allowed;
  color: #98A7B7;
  padding: 0 1.6rem;
}

.c4 {
  display: grid;
  grid-template-columns: repeat(1,1fr);
}

.c3 {
  white-space: initial;
  font-size: 1.4rem;
  line-height: 2.2rem;
  margin-top: 0.3rem;
  margin-bottom: 1.8rem;
  color: #98A7B7;
}

.c11 {
  position: relative;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  box-sizing: border-box;
  border: 0.15rem solid;
  width: 100%;
  height: 100%;
  border-radius: 100%;
}

.c11::after {
  content: '';
  position: absolute;
  top: 50%;
  left: 50%;
  width: 67%;
  height: 67%;
  -webkit-transform: translate(-50%,-50%) scale(0);
  -ms-transform: translate(-50%,-50%) scale(0);
  transform: translate(-50%,-50%) scale(0);
  border-radius: 100%;
  -webkit-transition: -webkit-transform 100ms ease-out;
  -webkit-transition: transform 100ms ease-out;
  transition: transform 100ms ease-out;
}

.c9 {
  position: absolute;
  opacity: 0;
  width: 1%;
  height: 1%;
  margin: 0;
}

.c9 ~ .c10 {
  border-color: #13181D;
}

.c9 ~ .c10::after {
  background-color: #13181D;
}

.c9:checked ~ .c10::after {
  -webkit-transform: translate(-50%,-50%) scale(1);
  -ms-transform: translate(-50%,-50%) scale(1);
  transform: translate(-50%,-50%) scale(1);
}

.c9:disabled ~ .c10 {
  border-color: #98A7B7;
}

.c9:disabled ~ .c10::after {
  background-color: #98A7B7;
}

.c9:not(:disabled):checked ~ .c10 {
  border-color: #3872D2;
}

.c9:not(:disabled):checked ~ .c10::after {
  background-color: #3872D2;
}

.c9:not(:disabled):focus ~ .c10 {
  box-shadow: 0 0 0.8rem 0 rgba(56,114,210,0.35);
}

.c8 {
  margin: 0.3rem;
  width: 1.8rem;
  height: 1.8rem;
  position: relative;
  border-radius: 100%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  cursor: not-allowed;
  padding: 0.8rem 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

<div>
  <div
    aria-describedby="gender-helper-text"
    class="c0"
    id="gender-wrapper"
    type="radio"
  >
    <span
      class="c1 c2"
      disabled=""
      id="gender-label"
      type="radio"
    >
      Gender
    </span>
    <span
      class="c3"
      disabled=""
      id="gender-helper-text"
      type="radio"
    >
      Helper Text
    </span>
    <div
      class="c4"
      id="gender-options"
    >
      <label
        class="c5"
        disabled=""
        for="Female-gender"
        id="Female-gender-wrapper"
        sizes="[object Object]"
      >
        <span
          class="c6 c7"
          disabled=""
          id="Female-gender-label"
          type="radio"
        >
          Female
        </span>
        <div
          class="c8"
        >
          <input
            class="c9"
            disabled=""
            id="Female-gender"
            name="gender"
            sizes="[object Object]"
            type="radio"
            value="female"
          />
          <div
            class="c10 c11"
          />
        </div>
      </label>
      <label
        class="c5"
        disabled=""
        for="Male-gender"
        id="Male-gender-wrapper"
        sizes="[object Object]"
      >
        <span
          class="c6 c7"
          disabled=""
          id="Male-gender-label"
          type="radio"
        >
          Male
        </span>
        <div
          class="c8"
        >
          <input
            class="c9"
            disabled=""
            id="Male-gender"
            name="gender"
            sizes="[object Object]"
            type="radio"
            value="male"
          />
          <div
            class="c10 c11"
          />
        </div>
      </label>
    </div>
  </div>
</div>
`;

exports[`Radio Group should render properly 1`] = `
.c1 {
  margin: 0;
  color: inherit;
  font-size: 1.6rem;
  font-weight: 600;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.6rem;
  text-align: initial;
  display: inline-block;
}

.c6 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
  display: inline-block;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: inherit;
  color: inherit;
}

.c7 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: inherit;
  color: inherit;
  padding: 0 1.6rem;
}

.c4 {
  display: grid;
  grid-template-columns: repeat(1,1fr);
}

.c3 {
  white-space: initial;
  font-size: 1.4rem;
  line-height: 2.2rem;
  margin-top: 0.3rem;
  margin-bottom: 1.8rem;
  color: #13181D;
}

.c11 {
  position: relative;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  box-sizing: border-box;
  border: 0.15rem solid;
  width: 100%;
  height: 100%;
  border-radius: 100%;
}

.c11::after {
  content: '';
  position: absolute;
  top: 50%;
  left: 50%;
  width: 67%;
  height: 67%;
  -webkit-transform: translate(-50%,-50%) scale(0);
  -ms-transform: translate(-50%,-50%) scale(0);
  transform: translate(-50%,-50%) scale(0);
  border-radius: 100%;
  -webkit-transition: -webkit-transform 100ms ease-out;
  -webkit-transition: transform 100ms ease-out;
  transition: transform 100ms ease-out;
}

.c9 {
  position: absolute;
  opacity: 0;
  width: 1%;
  height: 1%;
  margin: 0;
}

.c9 ~ .c10 {
  border-color: #13181D;
}

.c9 ~ .c10::after {
  background-color: #13181D;
}

.c9:checked ~ .c10::after {
  -webkit-transform: translate(-50%,-50%) scale(1);
  -ms-transform: translate(-50%,-50%) scale(1);
  transform: translate(-50%,-50%) scale(1);
}

.c9:disabled ~ .c10 {
  border-color: #98A7B7;
}

.c9:disabled ~ .c10::after {
  background-color: #98A7B7;
}

.c9:not(:disabled):checked ~ .c10 {
  border-color: #3872D2;
}

.c9:not(:disabled):checked ~ .c10::after {
  background-color: #3872D2;
}

.c9:not(:disabled):focus ~ .c10 {
  box-shadow: 0 0 0.8rem 0 rgba(56,114,210,0.35);
}

.c8 {
  margin: 0.3rem;
  width: 1.8rem;
  height: 1.8rem;
  position: relative;
  border-radius: 100%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  cursor: pointer;
  padding: 0.8rem 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

.c5:hover .c10 {
  border-color: #3872D2;
  box-shadow: 0 0 0.8rem 0 rgba(56,114,210,0.35);
}

.c5:active .c10 {
  border-color: #3061B3;
  box-shadow: 0 0 0.8rem 0 rgba(56,114,210,0.5);
}

<div>
  <div
    aria-describedby="gender-helper-text"
    class="c0"
    id="gender-wrapper"
    type="radio"
  >
    <span
      class="c1 c2"
      id="gender-label"
      type="radio"
    >
      Gender
    </span>
    <span
      class="c3"
      id="gender-helper-text"
      type="radio"
    >
      Helper Text
    </span>
    <div
      class="c4"
      id="gender-options"
    >
      <label
        class="c5"
        for="Female-gender"
        id="Female-gender-wrapper"
        sizes="[object Object]"
      >
        <span
          class="c6 c7"
          id="Female-gender-label"
          type="radio"
        >
          Female
        </span>
        <div
          class="c8"
        >
          <input
            class="c9"
            id="Female-gender"
            name="gender"
            sizes="[object Object]"
            type="radio"
            value="female"
          />
          <div
            class="c10 c11"
          />
        </div>
      </label>
      <label
        class="c5"
        for="Male-gender"
        id="Male-gender-wrapper"
        sizes="[object Object]"
      >
        <span
          class="c6 c7"
          id="Male-gender-label"
          type="radio"
        >
          Male
        </span>
        <div
          class="c8"
        >
          <input
            class="c9"
            id="Male-gender"
            name="gender"
            sizes="[object Object]"
            type="radio"
            value="male"
          />
          <div
            class="c10 c11"
          />
        </div>
      </label>
    </div>
  </div>
</div>
`;
